### PR TITLE
If the socket times out, try once more

### PR DESCRIPTION
**Pull request by cccs-kevin**

### CODE REVIEW

File: kaspersky.py [hunk 1]
```diff
@@ -53,7 +54,11 @@ def start(self) -> None:
 
     def execute(self, request: ServiceRequest) -> None:
         request.result = Result()
-        icap_result = self.icap.scan_data(request.file_contents, request.file_name)
+        try:
+            icap_result = self.icap.scan_data(request.file_contents, request.file_name)
+        except timeout:
+            # Try again, but only once!
+            icap_result = self.icap.scan_data(request.file_contents, request.file_name)
```
Comment (cccs-rs) [inline kaspersky.py:61]: Let it raise the error if it fails/times out twice?

Comment (cccs-kevin) [reply kaspersky.py:61]: Yes, similar to how MD handles connection errors (https://github.com/CybercentreCanada/assemblyline-service-metadefender/blob/992a4cc60a57dfcde353bcc93e26ea47d24b8c27/metadefender.py#L233), we could just raise a Recoverable error but then that has the potential to get stuff in an infinite loop

Comment (cccs-rs) [reply kaspersky.py:61]: Could adopt a "three strikes, you're out" approach by encapsulating inside a for and break after icap_result; exception could be caught and logged or passed and lets the loop continue until it completes?

Comment (cccs-kevin) [reply kaspersky.py:61]: sure! good thinking!

**Comment by cccs-kevin** [comment]
Closing since this is irrelevant after timeout was extended https://github.com/CybercentreCanada/assemblyline-v4-service/commit/e9cca8c81a4acc0b5f764dc23937d7922656f4f5

**Pull request closed without merging**